utility/src/main/java/org/oppia/util/data/AsyncDataSubscriptionManager.kt

```diff
@@ -0,0 +1,69 @@
+package org.oppia.util.data
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import org.oppia.util.threading.ConcurrentQueueMap
+import org.oppia.util.threading.dequeue
+import org.oppia.util.threading.enqueue
+import org.oppia.util.threading.getQueue
+import javax.inject.Inject
+import javax.inject.Singleton
+
+internal typealias ObserveAsyncChange = suspend () -> Unit
+
+// TODO(BenHenning): Move this to the Dagger graph rather than making it a singleton.
```

Comment:
Here and elsewhere (e.g. HomeActivityTest.kt): please link TODOs to issue numbers.

Reply:
Your comment came before my commit to actually do this. :) Done.